_posts/2019-12-20-ar-signpost.md

```diff
@@ -0,0 +1,254 @@
+</a>
+</div>
+
+
+Let's put your favorite mall on the scene for a moment.
+In this mall you would find markers with QR codes strategically placed so that they are visible, and you have easy access to them.
+Each QR contains contextual information such as a unique identifier, the building in which it is located and the floor in which it is located.
+
+Imagine you are looking for a particular shop, the first step would be to scan the marker closest to you with the mobile application.
+This would send the information contained in the QR to our REST API and it would take care of obtaining the necessary information from the database by querying the information obtained from the QR.
+This information will be sent back to the mobile application, which contains all the points of interest around you, such as shops, escalators and much more.
+Now it's time for the funniest part of the process, to visualize these points of interest with AR as shown in the following video.
+
+<div style="position: relative; width: 100%; height: 0; padding-bottom: 55%;">
+    <iframe src="https://www.youtube.com/embed/Gmcl8uDneng" width="100%" height="100%;" style="position: absolute; left: 0; top: 0; bottom: 0; right: 0;"></iframe>
+</div>
+
+Are you wondering how this happens? Well this process is carried out with the ViroReact platform.
+First of all, we must take into account that the AR world is three-dimensional, so we have x, z and y-axis as shown in the image below.
+
+<div style="text-align: center; margin: 0px auto; width: 50%">
+ <a href="{{ '/img/2019-12-20-ar-signpost/x-y-z-axis.png' | prepend: site.baseurl }}" data-lightbox="ui" >
+<img alt="ikea space" src="{{ '/img/2019-12-20-ar-signpost/x-y-z-axis.png' | prepend: site.baseurl }}" class="image fit" style="margin:0px auto; max-width: 325px;" style="background-color: white">
+</a>
+</div>
+
+The AR scene starts from the point where the camera is located, which is usually [x=0, y=0, z=0] and the objects are located around it with different coordinates.
+In this case we are going to focus only on x and z-axis because y-axis represents the height of the objects so it will be constant.
+
+The points of interest we receive contain geographic coordinates (latitude and longitude) where they are located in the real world.
+The AR world does not understand geographic coordinates, so we have to convert these GPS coordinates to point x, z and y-axis in meters taking as reference the initial position of the AR camera.
+For this we use a technique called Web Mercator projection and in this way, we have our points of interest integrated into the real world as shown in the video above.
+
+Going back to our example before the mall, you wondered how this system would be implemented in a mall, didn't you? 
+Well there the web application intended for administrators or operators who are responsible for indicating where each point of interest is located in the building comes into play.
```

Comment:
Well there, the web ... (comma makes it more readable I think)

Reply:
> Great post! Just a few small remarks

Thank you very much for your feedback 🙂